File: boats-to-save/go_lang/solution_test.go
package go_lang

import "testing"

func TestBoatsToSave01(t *testing.T) {
	want := 3
	limit := 3
	people_weights := []int{3, 2, 1, 2}
	got := BoatsToSave(people_weights, limit)
	if got != want {
		t.Errorf("got %v want %v", got, want)
	}
}

func TestBoatsToSave02(t *testing.T) {
	want := 4
	limit := 5
	people_weights := []int{3, 5, 3, 4}
	got := BoatsToSave(people_weights, limit)
	if got != want {
		t.Errorf("got %v want %v", got, want)
	}
}

func TestBoatsToSave03(t *testing.T) {
	want := 1
	limit := 3
	people_weights := []int{1, 2}
	got := BoatsToSave(people_weights, limit)
	if got != want {
		t.Errorf("got %v want %v", got, want)
	}
}

File: boats-to-save/python/solution_test.py
import unittest

from solution import Solution

class SolutionTest(unittest.TestCase):
    
    def test_boats_to_save_01(self):
        people_weights = [3,2,1,2]
        boat_weight_limit = 3
        sol = Solution()
        no_of_boats_required = sol.boats_to_save(people_weights, boat_weight_limit)
        self.assertEqual(3, no_of_boats_required)

    def test_boats_to_save_02(self):
        people_weights = [3,5,3,4]
        boat_weight_limit = 5
        sol = Solution()
        no_of_boats_required = sol.boats_to_save(people_weights, boat_weight_limit)
        self.assertEqual(4, no_of_boats_required)

    def test_boats_to_save_03(self):
        people_weights = [1,2]
        boat_weight_limit = 3
        sol = Solution()
        no_of_boats_required = sol.boats_to_save(people_weights, boat_weight_limit)
        self.assertEqual(1, no_of_boats_required)

if __name__ == "__main__":
    unittest.main()